### DISCUSSION TITLE
FreeTDS: db-lib: exiting because client error handler returned INT_EXIT for msgno 20109

### DISCUSSION THREAD

**Discussion by JamieMellway** [FreeTDS/freetds#428] [Q&A]
I am calling a binary on RedHat 7 that uses the FreeTDS library. It works fine in one environment set up long ago, but I'm trying to run the binary in other environments and I keep getting `FreeTDS: db-lib: exiting because client error handler returned INT_EXIT for msgno 20109.`  I also tried this on an Ubuntu machine and I get the same error.

I've tried just copying the lib folder, copying the all of the files, and also downloading the files and doing configure/make/makeinstall. All those ways give the same error.

The FreeTDS lib folder is listed in the LD_LIBRARY_PATH environment variable.

When I look in the freetds code, I see:
```
#define SYBENULL 20109 /* NULL DBPROCESS pointer passed to DB-Library. */
#define CHECK_PARAMETER(x, msg, ret) if (!(x)) { dbperror(dbproc, (msg), 0); return ret; }

#CHECK_PARAMETER(dbproc, SYBENULL, FALSE);
```
That parameter check on dbproc is in many places in the freetds code.

How do I setup FreeTDS to avoid this error?

https://serverfault.com/questions/1081411/freetds-returned-int-exit-for-msgno-20109

**Comment by fziglio** [comment]
The question is more "is this the standard proprietary behaviour or not?"
